### PR TITLE
Add new Persistence Diagram backend

### PR DESCRIPTION
**Pull request by pierre-guillou**
This PR adds a new Persistence Diagram backend named "Persistent Simplex Pairs". This algorithm is described in "Algorithm and Theory of Computation Handbook (Second Edition) - Special Topics and Techniques" by Atallah and Blanton on page 97.

Since I used the `VisitedMask` data structure from the `DiscreteGradient` module, this structure has been moved in the `common` module.

This algorithm is very slow - 4 to 5 orders of magnitudes slower than FTM. However, it computes saddle-saddle and infinite pairs.

Enjoy,
Pierre

### CODE REVIEW

File: core/base/common/VisitedMask.h [hunk 1]
```diff
@@ -0,0 +1,59 @@
+#pragma once
```
Comment (julien-tierny) [inline core/base/common/VisitedMask.h:6]: thanks for adding your name, email address and date.

File: paraview/xmls/PersistenceDiagram.xml [hunk 2]
```diff
@@ -1,67 +1,72 @@
...
+        Persistence diagrams are useful and stable concise representations of the
+        topological features of a data-set. It is useful to fine-tune persistence
+        thresholds for topological simplification or for fast similarity
+        estimations for instance.
 
-Persistence diagrams are useful and stable concise representations of the
-topological features of a data-set. It is useful to fine-tune persistence
-thresholds for topological simplification or for fast similarity
-estimations for instance.
+        Two backends are available for the computation of the persistence diagram:
```
Comment (julien-tierny) [inline paraview/xmls/PersistenceDiagram.xml]: Two --› Three

**Review by julien-tierny — CHANGES_REQUESTED** [review]
Thanks for this PR
Could you address my couple of remarks?
Thanks

File: core/base/common/VisitedMask.h (hunk 1 above)
Comment (pierre-guillou) [reply core/base/common/VisitedMask.h:6]: Fixed

File: paraview/xmls/PersistenceDiagram.xml (hunk 2 above)
Comment (pierre-guillou) [reply paraview/xmls/PersistenceDiagram.xml]: Fixed

**Review by julien-tierny — APPROVED** [review]

**Comment by julien-tierny** [comment]
For some reason, the Azure Windows CI is now exceeding its time limit and fails all the time.
As it seems to be redundant with the `test-build-windows` github action, I'd suggest to remove the Azure Windows CI.
What do you think?

**Comment by julien-tierny** [comment]
awesome, thanks!

**Pull request merged**